### PR TITLE
New method removePermission() for AuthManager

### PR DESCRIPTION
**Pull request by DanHarrin**
One of my plugins requires the programatic removal of a permission, so I added a method for `AuthManager` called `removePermission()`. It can be used like so:

```
BackendAuth::registerCallback(function ($manager) {
    $manager->deregisterPermission('October.Cms', 'cms.manage_content');
});
```

### CODE REVIEW

**Comment by w20k** [comment]
Just a thought, might be useful - to add a method for User `Impeachment` (😄) to remove all User Permissions in one go :)

**Comment by DanHarrin** [comment]
@w20k this is for hiding permissions in the list, not for removing permissions from a user :)

**Comment by DanHarrin** [comment]
I've renamed this method to `deregisterPermission()` instead, to make it super clear 👍

**Comment by DanHarrin** [comment]
Sorry, this doesn't actually work at the moment 😂 just debugging

**Comment by DanHarrin** [comment]
It's actually working now @w20k haha 😂

**Comment by LukeTowers** [comment]
@DanHarrin @w20k I prefer removePermission(), the method exists on the AuthManager class, not the User class so I think it's fine as removePermission().

**Comment by DanHarrin** [comment]
@LukeTowers all done :+1:

File: modules/backend/classes/AuthManager.php [hunk 1]
```diff
@@ -96,6 +96,26 @@ public function registerPermissions($owner, array $definitions)
         }
     }
 
+    /**
+     * Removes a single back-end permission
```
Comment (bennothommo) [inline modules/backend/classes/AuthManager.php]: @DanHarrin Would you mind adding the parameters and return value to the docblock?
```suggestion
     * Removes a single back-end permission
     * @param string $owner Specifies the menu item's owner plugin or module in the format Vendor/Module.
     * @param string $code The code of the permission to remove
     * @return void
```

**Comment by DanHarrin** [comment]
That's done too, @bennothommo 👍

**Comment by LukeTowers** [comment]
Related: https://github.com/octobercms/october/issues/3559

File: modules/backend/classes/AuthManager.php [hunk 2]
```diff
@@ -96,6 +96,29 @@ public function registerPermissions($owner, array $definitions)
...
+     * @param string $owner Specifies the menu item's owner plugin or module in the format Vendor/Module.
```
Comment (LukeTowers) [inline modules/backend/classes/AuthManager.php]: Can you fix this docblock (and the one for registerPermissions) to say `Specifies the permissions' owner plugin or module in the format Author.Plugin`

**Comment by DanHarrin** [comment]
Docblocks have been updated @LukeTowers 👍

**Comment by bennothommo** [comment]
@DanHarrin @LukeTowers I've added some test cases for `AuthManager`, which should test this functionality along with registering permissions. I've also tested this in normal operation and it all works well.

**Comment by LukeTowers** [comment]
@DanHarrin can you add this to the docs?

**Comment by DanHarrin** [comment]
Sure thing @LukeTowers, all my other PRs need documenting as well so I'll get them all done when I find some time.

**Pull request merged**